File: src/java/uk/co/bwhazel/jcounterpoise/dataformatters/IDataFormatter.java
/**
 * Copyright: (c) Benedict W. Hazel, 2011-2012
 * IDataFormatter: Interface implemented by data exporting classes.
 */

package uk.co.bwhazel.jcounterpoise.dataformatters;

import java.io.IOException;
import java.io.OutputStream;
import uk.co.bwhazel.jcounterpoise.IEncounter;

/**
 * Defines methods implemented by data exporting classes.
 * @author Benedict
 */
public interface IDataFormatter {
    /**
     * Exports the counterpoise correction calculation data in the format
     * supported by the class.
     * @param encounter Instance of class implementing IEncounter containing the
     * calculation data.
     * @param stream Stream to write formatted data into.
     * @throws IOException Exception thrown if an error occurs while exporting
     * data.
     */
    void exportData(IEncounter encounter, OutputStream stream)
            throws IOException;
}

File: 1.0-northcote/cs/Sharpen/DataFormatters/IDataFormatter.cs
// <copyright file="IDataFormatter.cs" company="Benedict W. Hazel">
//      Benedict W. Hazel, 2011-2012
// </copyright>
// <author>Benedict W. Hazel</author>
// <summary>
//      IDataFormatter: Interface implemented by data exporting classes.
// </summary>

using System.IO;

namespace BWHazel.Sharpen.DataFormatters
{
    /// <summary>
    /// Defines methods implemented by data exporting classes.
    /// </summary>
    public interface IDataFormatter
    {
        /// <summary>
        /// Exports the counterpoise correction calculation data in the format supported by the class.
        /// </summary>
        /// <param name="encounter">Instance of class implementing <see cref="IEncounter"/> containing the calculation data.</param>
        /// <param name="stream">Stream to write formatted data into.</param>
        void ExportData(IEncounter encounter, Stream stream);
    }
}
